src/main/java/com/gradle/CustomBuildScanEnhancements.java

```diff
@@ -395,15 +398,17 @@ public void execute(BuildResultAdapter buildResult) {
 
     private void captureGitMetadata() {
         // Run expensive computation in background
-        buildScan.background(new CaptureGitMetadataAction(providers, develocity));
+        buildScan.background(new CaptureGitMetadataAction(projectDir, providers, develocity));
     }
 
     private static final class CaptureGitMetadataAction implements Action<BuildScanAdapter> {
 
         private final ProviderFactory providers;
         private final DevelocityAdapter develocity;
+        private final File projectDir;
 
-        private CaptureGitMetadataAction(ProviderFactory providers, DevelocityAdapter develocity) {
+        private CaptureGitMetadataAction(File projectDir, ProviderFactory providers, DevelocityAdapter develocity) {
```

Comment:
I feel aligning the argument order with the constructor of `CustomBuildScanEnhancements` would help to read the code. The ordering was unaligned before your change though.